### PR TITLE
fix: apply initialData appState for zenmode and grid stats and refactor check param for actions

### LINKED ISSUE #2870: Context menu actions broken when set from initialData
In #2659 , the context menu options have been changed to actions and each action has a default value of `checked` set as `false` hence initialData appState is not getting reflected.

**Steps to reproduce**
1. Open [excalidraw](https://excalidraw.com/) and switch to zen mode / grid mode
2. Refresh and zen mode/grid mode is rendered
3. Use the context menu to toggle zen mode/grid mode. It will not toggle for the first time since the default value is false.

### PR DESCRIPTION
**Pull request by ad1992**
fixes #2870

### CODE REVIEW

**Comment by dwelle** [comment]
I'm facepalming myself — didn't think of state persistence.

That said, this fix won't do. Mixing type objects and functions returning type objects is introducing several problems: 1) it's not pretty 2) it's not performant (having to rebuild/register actions multiple times) 3) and since we don't type check the exported actions, it will be error prone. In fact, it already is: it breaks shortcut handling.

I propose we make `checked` from `boolean` into `() => boolean` (or at least `(boolean | () => boolean)`.

File: src/actions/actionToggleGridMode.tsx [hunk 1]
```diff
@@ -2,20 +2,21 @@ import { CODES, KEYS } from "../keys";
...
+        appState: {
+          ...appState,
+          gridSize: this.checked ? GRID_SIZE : null,
+        },
+        commitToHistory: false,
+      };
+    },
+    checked,
+    contextItemLabel: "labels.gridMode",
+    // Wrong event code
```
Comment (lipis) [inline src/actions/actionToggleGridMode.tsx:20]: What's this?

File: src/actions/actionToggleZenMode.tsx [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
+        appState: {
+          ...appState,
+          zenModeEnabled: this.checked,
+        },
+        commitToHistory: false,
+      };
+    },
+    checked,
+    contextItemLabel: "buttons.zenMode",
+    // Wrong event code
```
Comment (lipis) [inline src/actions/actionToggleZenMode.tsx:19]: and this.. :) can we remove it or put the _right_ code?

Comment (ad1992) [reply src/actions/actionToggleZenMode.tsx]: no idea!  but fixed and it didn't get caught initially since it wasn't getting used :D

**Review by dwelle — APPROVED** [review]
changelog?

**Comment by ad1992** [comment]
> changelog?

@dwelle since we will anyways update the changelog before we do the next version release so it will be easier to add at that time since this not an API change so pr title can be used as it is in changelog. (I pull all commits pushed after the last release and categorize it, will soon automate the categorizing then it will be easy to generate changelog).

**Pull request merged**